### ISSUE TITLE
Best way to concatenate jrnls?

### ISSUE THREAD

**Issue by Serif-7** [jrnl-org/jrnl#1292]
Let's say I make a new jrnl and want to import older entries. Is the best way to do this just `cat jrnl1 jrnl2 > jrnl_main.txt`? I notice there's no break between the old and new entries. Since it's just text, I can just edit it and add it in right? The parsing doesn't do any weird stuff that might mess that up?

**Comment by viegasfh** [comment]
Yes, it should work as long as you have the journal structure

[date time] Title goes here
Body

That's the format for each entry as I have found. Now, if you have defined XML, JSON or Markdown, then it has a different structure. Experiment with the --export parameter and you should learn more about it.

**Comment by Serif-7** [comment]
I lost some entries after using my command... note to anybody finding this issue, back up your jrnls.

**Comment by viegasfh** [comment]
You should have used >> to concatenate. > wipes the existing file.

**Comment by wren** [comment]
jrnl has an import command for this. So, you can pipe the entire file you are trying to concatenate to `jrnl --import` and it will do what you expect (e.g. `cat myfile.txt | jrnl --import`). The advantage of this approach over concatenating the journal files directly is that jrnl will de-dupe incoming entries for you.

That's fine if you want to import from a file, but as you mention above, you're trying to concatenate from one journal to another one. So, you can use jrnl for that, too. The default format for jrnl is meant for display in your terminal, so it's not suitable for this purpose. Luckily, jrnl supports a [bunch of formats](https://jrnl.sh/en/stable/formats/), including the plain text format used to store your journal file (see "[Plain Text](https://jrnl.sh/en/stable/formats/#plain-text)" on the formats page).

Long story short, you can do something like this (where `one` is the journal you're getting entries from, and `two` is the journal you're concatenating to):
```bash
jrnl one --format text | jrnl two --import
```

And the advantage of this is that you can optionally search for entries the same way you would with any other jrnl command. So, any of these would also work:

```bash
jrnl one --format text -from saturday -to today | jrnl two --import
jrnl one --format text -contains "some text" | jrnl two --import
jrnl one --format text -starred | jrnl two --import
```

And so on. Any search you can normally do with jrnl can be used like this just by adding `--format text`. And as I mentioned above, the entries are de-duped as they're imported, so you can do the same command multiple times without spamming the `two` journal with a bunch of duplicate entries.

Does that all make sense?

**Comment by viegasfh** [comment]
Hi again, @wren!

The import feature also has a bug. If you are working with single file journals, the import works perfectly. But, if your target journal is a directory it just says that the files were imported to the default journal, but nothing gets really imported.

To reproduce this, you can do the following:

1. Create a folder for your journal
2. Add your folder journal to the config file
3. Now run the import command

What I did was this:

```
jrnl personal --export text | jrnl two --import
```

and this is what I got as response:

```
[15 imported to default journal]
```

However, in reality nothing was imported.

If I change my two journal to a single file, then I get it work:

```
jrnl personal --export text | jrnl two --import
```

and output is:

```
[Journal 'two' created at /home/user01/.local/share/jrnl/two.txt]
[15 imported to two journal]
```

In this case, because the file didn't exist, it created it first and the imported the entries without an issue.

It would be nice to be able to import the entries to a directory journal.

**Comment by wren** [comment]
@viegasfh I see you've run into #999.

In the past our test suite didn't really have any coverage for storage formats other than the default "onefile" format. We've been updating our testing to have full coverage for all storage formats, and are finding that some of the other formats (especially the directory-based format) have some bugs we never knew about. 

Sorry about that! We're prioritizing bug fixes soon after we finish the testing changes, since at that point we'll actually know everything we need to fix.

**Comment by viegasfh** [comment]
Hi, @wren !

Yes, I did. Thanks for noticing!

Don't worry about the bugs. I believe not many people bothered with the directory journals. I like it, because that's how I have been doing it by hand. In addition that's what was making me choose between Journal-Cli and Jrnl. Initially I began with Journal-Cli, but I then opted for the single file. When looking into the code, I was quite surprised that Jrnl did far more than I thought it did, particularly the directoy journals. So, I decided to experiment with it and, hey, I got into the bugs. But that's not too much of a deal at this moment. As soon as you fix them, I will migrate to the directory based journals.

Thanks for all the effort!
